pkg/codegen/dotnet/gen.go

```diff
@@ -1380,7 +1380,7 @@ func (mod *modContext) genFunction(w io.Writer, fun *schema.Function) error {
 	// Emit the datasource method.
 	fmt.Fprintf(w, "        public static Task%s InvokeAsync(%sInvokeOptions? options = null)\n",
```

Comment:
Should we be fully qualifying other types like `InvokeOptions` as `global::Pulumi.InvokeOptions`? (There are probably other similar cases).

Reply:
I almost wonder if we should be fully qualifying _everything_, even types from the `System` namespace. It'll look ugly, but reduce the chances of conflicts. Maybe we don't need to go that far just yet.

Reply:
> Maybe we don't need to go that far just yet.

I think the problem arises when when we a fully qualified type such as `Pulumi.Config` that is not marked with `global::`. Other Pulumi types we use are not fully qualified: `Input<T>`, `Output<T>`, `InvokeOptions` and these resolve correctly. I think it is unlikely that we get a type collision with these from the generated types of the schema